Add TrainingDataWithPremises Extraction

JOSHCLUNE opened the pull request.

wellecks commented on scripts/run_pipeline.py:
```diff
@@ -33,32 +35,44 @@ def _run_cmd(cmd, cwd, input_file, output_file):
-            _get_stem(input_module, input_file_mode) + '.jsonl'
-        )
-    )
+    # _run_cmd(
```
Perhaps we can add a command line flag to determine which of these is run (rather than commenting out)

wellecks reviewed (changes requested):
Left a comment about a command-line option for `run_pipeline.py`. Otherwise this looks great, thanks!

wellecks commented:
Also, could you add a short description of the new commands to the readme?

JOSHCLUNE commented:
I've added command-line options for run_pipeline and updated the README to explain them in the most recent commit. Further additions (such as adding a field for all premises used for the current declId proof or a field for all premises from the current state to the end of the goal) will be handled in a later PR once I am able to address them. Let me know if there are any additional issues, thanks.

wellecks commented:
Awesome, thanks!
